src/ZiggyCreatures.FusionCache.Backplane.NATS/NatsBackplane.cs

```diff
@@ -0,0 +1,142 @@
+﻿using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Abstractions;
+
+using NATS.Client.Core;
+
+namespace ZiggyCreatures.Caching.Fusion.Backplane.NATS;
+
+/// <summary>
+/// A NATS based implementation of a FusionCache backplane.
+/// </summary>
+public partial class NatsBackplane
+	: IFusionCacheBackplane
+{
+	private BackplaneSubscriptionOptions? _subscriptionOptions;
+	private readonly ILogger? _logger;
+	private INatsConnection _connection;
+	private string _channelName = "";
+	private Func<BackplaneMessage, ValueTask>? _incomingMessageHandlerAsync;
+	private INatsSub<NatsMemoryOwner<byte>>? _subscription;
+
+	/// <summary>
+	/// Initializes a new instance of the NatsBackplane class.
+	/// </summary>
+	/// <param name="natsConnection">The NATS connection instance to use.</param>
+	/// <param name="logger">The <see cref="ILogger{TCategoryName}"/> instance to use. If null, logging will be completely disabled.</param>
+	public NatsBackplane(INatsConnection? natsConnection, ILogger<NatsBackplane>? logger = null)
```

Comment:
It's a bad idea to make a parameter nullable if you require this service (NATS connection). You don't need to make it nullable to check it for `null` and throw ANE in code. Nullable reference types good only for static analysis and in code you should always check argument for `null` anyway.